unittests/allplatformstests.py

```diff
@@ -833,6 +833,26 @@ def test_build_by_default(self):
         self.build(target=('barprog' + exe_suffix))
         self.assertPathExists(exe2)
 
+    def test_build_generated_pyx_directly(self):
+        # Check that the transpile stage also includes
+        # dependencies for the compilation stage as dependencies
+        testdir = os.path.join("test cases/cython", '2 generated sources')
+        env = get_fake_env(testdir, self.builddir, self.prefix)
+        try:
+            detect_compiler_for(env, "cython", MachineChoice.HOST)
+        except EnvironmentException:
+            raise SkipTest("Cython is not installed")
+        self.init(testdir)
+        # Need to get the full target name ( can't know earlier since it includes stuff like Python version )
+        targets = self.introspect('--targets')
+        target = next(filter(lambda t: "includestuff.cpython" in t["name"], targets))  # Only will be one target
+        # Meson introspection doesn't give us the pyx.c target :(
+        # This is really hacky
+        name = target["target_sources"][0]["generated_sources"][0]
```

Comment:
I guess in theory you could also check to see which targets have a target_sources -> generated_sources ending in "includestuff.pyx.c"?

Reply:
Good point. I've updated this.